app/src/main/java/com/asfoundation/wallet/nfts/ui/nftTransactDialog/NFTTransactDialogFragment.kt

```diff
@@ -0,0 +1,108 @@
+package com.asfoundation.wallet.nfts.ui.nftTransactDialog
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.asf.wallet.R
+import com.asf.wallet.databinding.FragmentNftTransactBinding
+import com.asfoundation.wallet.base.Async
+import com.asfoundation.wallet.base.SingleStateFragment
+import com.asfoundation.wallet.di.DaggerBottomSheetDialogFragment
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import java.math.BigInteger
+import javax.inject.Inject
+
+class NFTTransactDialogFragment : DaggerBottomSheetDialogFragment(),
+    SingleStateFragment<NFTTransactState, NFTTransactSideEffect> {
+
+  @Inject
+  lateinit var viewModelFactory: NFTTransactDialogViewModelFactory
+
+  private val viewModel: NFTTransactDialogViewModel by viewModels { viewModelFactory }
+  private val views by viewBinding(FragmentNftTransactBinding::bind)
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    return inflater.inflate(R.layout.fragment_nft_transact, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    viewModel.collectStateAndEvents(lifecycle, viewLifecycleOwner.lifecycleScope)
+    setListeners()
+  }
+
+  override fun onStart() {
+    val behavior = BottomSheetBehavior.from(requireView().parent as View)
+    behavior.state = BottomSheetBehavior.STATE_EXPANDED
+    super.onStart()
+  }
+
+  override fun getTheme(): Int = R.style.AppBottomSheetDialogTheme
+
+  override fun onStateChanged(state: NFTTransactState) {
+    views.layoutNftTransactDone.doneMensage.text =
+        getString(R.string.nfts_transact_done_mensage, state.data.name)
+
+    when (val transactionHash = state.transactionHashAsync) {
+      is Async.Fail -> showError(getString(R.string.nfts_generic_error))
+      is Async.Loading -> showLoading()
+      is Async.Success -> if (transactionHash.value!!.startsWith(
```

Comment:
When working with nullable variables such as this transactionHash.value it’s a good idea to use “?” or “?:” instead of “!!” for null safety. Here, if transactionHash.value!! is null then it would throw an exception.